component/src/main/ballerina/ballerina/net/reliable/httpr/httpr.bal

```diff
@@ -0,0 +1,344 @@
+import ballerina.net.reliable.storejms;
+
+public const string MESSAGE_STORE_DESTINATION_NAME = "_BallerinaHTTPMessageStore_";
+
+@Description { value:"Http client connector for reliable outbound HTTP requests"}
+@Param { value:"serviceUri: Url of the service" }
+@Param { value:"connectorOptions: connector options" }
+@Param { value:"guaranteedProcessor: processor which responsible for handling guaranteed processing" }
+public connector HttpGuaranteedClient (string serviceUri, http:Options connectorOptions, processor:GuaranteedProcessor guaranteedProcessor) {
+
+    @Description { value:"GET action implementation of the reliable HTTP Connector"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action get (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "get") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"POST action implementation of the reliable HTTP Connector"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action post (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "post") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"HEAD action implementation of the reliable HTTP Connector"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action head (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "head") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"PUT action implementation of the reliable HTTP Connector"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action put (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "put") ;
```

Comment:
can we use an enum instead of "put", "patch" etc?

Reply:
Issue is, we have are supporting custom http methods using execute method, (which takes http method as a string)